OrbitSsh/include/OrbitSsh/DirectTcpChannelManager.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_SSH_DIRECT_TCP_CHANNEL_MANAGER_H_
+#define ORBIT_SSH_DIRECT_TCP_CHANNEL_MANAGER_H_
+
+#include <libssh2.h>
+
+#include <memory>
+
+#include "Channel.h"
+#include "ResultType.h"
+
+namespace OrbitSsh {
+
+class DirectTcpChannelManager {
+ public:
+  enum State {
+    kNotInitialized,
+    kRunning,
+    kSentEOFToRemote,
+    kRemoteSentEOFBack,
+    kClosed
+  };
+
+  DirectTcpChannelManager(Session* session_ptr, int remote_port,
+                          std::string remote_host = "localhost");
+  DirectTcpChannelManager(const DirectTcpChannelManager&) = delete;
+  DirectTcpChannelManager& operator=(const DirectTcpChannelManager&) = delete;
+  DirectTcpChannelManager(DirectTcpChannelManager&&) = default;
```

Comment:
You probably meant delete here, right?

Reply:
No, it is used for moving. This specifically originates from having std::vector<TunnelManager> in Ssh Manager. Resizing the vector means moving the object to a different location. And DirectTcpChannel is a member in TunnelManager.

Reply:
Makes sense. So move-assignment should be possible as well, right?

Reply:
i guess, done